civicrm.user.inc

```diff
@@ -24,6 +24,13 @@ function civicrm_user_insert(AccountInterface $account) {
   }
   \CRM_Core_BAO_UFMatch::updateContactEmail($contact_id, $account->getEmail());
 
+  // If already in CiviCRM, we don't do then next bit because it will cause
+  // the current request to exit, stopping execution.
+  $config = CRM_Core_Config::singleton();
+  if ($config->inCiviCRM) {
+    return;
```

Comment:
The Drupal 7 module does a similar thing. See https://github.com/civicrm/civicrm-drupal/blob/7.x-master/civicrm_user.inc#L64